src/core/Akka/Actor/FSM.cs

```diff
@@ -750,7 +752,7 @@ public void StartWith(TState stateName, TData stateData, TimeSpan? timeout = nul
         /// <returns>Descriptor for staying in the current state.</returns>
         public State<TState, TData> Stay()
         {
-            return GoTo(_currentState.StateName).WithNotification(false);
+            return _currentState.WithNotification(false);
```

Comment:
Biggest memory savings - don't allocate a new `State` object each time we call `Stay`. Re-use the same value.

Reply:
FWIW, there's no benefit to copying the same state over and over again here. The `Reason`, `Replies` properties et al don't have any effects when used here in combination with `WithNotification(false)`.